### ISSUE TITLE
[Azure.Core] Add Public Model Serialization 

### ISSUE THREAD

**Issue by nisha-bhatia** [Azure/azure-sdk-for-net#35665]
Add Public Model Serialization classes to all Models
To be completed by May Milestone

Checklist:
- [ ] Add SerializableOptions as a parameter to IUtf8JsonSerializable
- [ ] https://github.com/Azure/azure-sdk-for-net/blob/652dc91766f8336af2d9e36604d29a9f782fe4e4/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Utf8JsonWriterExtensions.cs#L18 needs case for ISerializable

Fixes:

- https://github.com/Azure/azure-sdk-for-net/issues/37094
- https://github.com/Azure/azure-sdk-for-net/issues/35783
- https://github.com/Azure/azure-sdk-for-net/issues/28358
- https://github.com/Azure/azure-sdk-for-net/issues/35201#issuecomment-1503756381
- https://github.com/Azure/azure-sdk-for-net/issues/35963
- https://github.com/Azure/autorest.csharp/issues/3353
- https://github.com/Azure/azure-sdk-for-net/issues/37475

**Comment by nisha-bhatia** [comment]
https://github.com/Azure/azure-sdk-for-net/issues/35783

**Comment by nisha-bhatia** [comment]
https://github.com/Azure/azure-sdk-for-net/issues/28358

**Comment by nisha-bhatia** [comment]
https://github.com/Azure/azure-sdk-for-net/issues/35201#issuecomment-1503756381

**Comment by nisha-bhatia** [comment]
https://github.com/Azure/azure-sdk-for-net/issues/35963

**Comment by nisha-bhatia** [comment]
https://github.com/Azure/autorest.csharp/issues/3353

**Comment by jsquire** [comment]
Diplicate: #38763

**Comment by weng5e** [comment]
Thanks for pointing to this tracking work item from the duplicated issues. It seems like the mile stone has been changed several times. Is this work item commited? Could you please also share an very rough ETA for this work item?

**Comment by jsquire** [comment]
The only statement that I can make is that this is something that we are working on.  There is no committed timeline that I can share.

**Comment by jsquire** [comment]
Duplicate: #39107.

**Comment by BlackGad** [comment]
Could you include clone methods? In our use cases, we need to adjust request options just before dispatching them. Modifying properties of the original instance is not convenient operation :)

**Comment by annelo-msft** [comment]
@m-nash, I believe the work on the Core side of this is complete with the release of System.ClientModel v1.0.0.  If work remains to implement model serialization in various client libraries, should this be moved to track those separately?

**Comment by stevendarby** [comment]
Find the tracking of this work very hard to follow.

Items raised for specific models are closed as duplicates, referring to this item, e.g. https://github.com/Azure/azure-sdk-for-net/issues/38763#issuecomment-1723351506

But comment above suggests the core work is done and individual clients/models may need to add support, so should some of those "duplicates" be re-opened?

Also the issue lists several issues which it is supposed to fix when done, but some of those are already closed, leaving the current state of this work unclear. 

Personally I'm interested the status of AnalyzeResult.
